### ISSUE TITLE
update for julia 0.5

### ISSUE THREAD

**Issue by CarloLucibello** [jpfairbanks/GraphMatrices.jl#16]
I'm trying to update LightGraphs to julia 0.5 but I get some failures due to this package. Can you fix that @jpfairbanks?

Thanks,
Carlo

**Comment by sbromberger** [comment]
https://travis-ci.org/JuliaGraphs/LightGraphs.jl/jobs/147669854#L4631

**Comment by jpfairbanks** [comment]
Yes I can fix it. I have a lot going on right now but will get to it soon. What is the failure on 0.5? I have not switched to 0.5 for my main work yet.

**Comment by jpfairbanks** [comment]
I just pushed a branch for 0.5 support see PR #17.
The breakage is caused by the fact that `Base.Linalg.eigs` uses `A_mul_B!` instead of `*`.
Which means all of my types need to define that function.

Do you think that Compat will be able to make that work around that? 
Having both definitions should provide support for both.

 Warnings were introduced to use `Diagonal(x)*y` instead of `scale!(x, y), so the fix for that probably breaks 0.4 support anyway.
